apps/src/dance/cdoFieldDanceAi.js

```diff
@@ -0,0 +1,164 @@
+import GoogleBlockly from 'blockly/core';
+import experiments from '@cdo/apps/util/experiments';
+import color from '@cdo/apps/util/color';
+import {getStore} from '../redux';
+import {setCurrentAiModalField} from './danceRedux';
+
+const FIELD_WIDTH = 32;
+const FIELD_HEIGHT = 18;
+const FIELD_PADDING = 2;
+
+class CdoFieldDanceAi extends GoogleBlockly.Field {
+  constructor(options) {
+    super();
+
+    this.options = options;
+    this.SERIALIZABLE = true;
+    this.CURSOR = 'default';
+    this.backgroundElement = null;
+  }
+
+  saveState() {
+    return this.getValue();
+  }
+
+  loadState(state) {
+    this.setValue(state);
+  }
+
+  static fromJson(options) {
+    return new CdoFieldDanceAi(options);
+  }
+
+  initView() {
+    this.createBorderRect_();
+    this.createTextElement_();
+    if (this.borderRect_) {
+      this.borderRect_.classList.add('blocklyDropdownRect');
+    }
+
+    this.backgroundElement = GoogleBlockly.utils.dom.createSvgElement(
+      'g',
+      {
+        transform: 'translate(1,1)',
+      },
+      this.fieldGroup_
+    );
+
+    this.updateSize_();
+  }
+
+  applyColour() {
+    const style = this.sourceBlock_.style;
```

Comment:
Just noting a question here for next iteration - do we want the color of the block to change once the stage has been generated and used?
I know the slots are 'TBD' but curious about the color.

Reply:
Let's discuss during the next round of design!